scripts/deploy-sc.sh

```diff
@@ -34,6 +34,6 @@ echo "Installing helm charts" >&2
 cd "${SCRIPTS_PATH}/../helmfile"
 declare -a helmfile_opt_flags
 [[ -n "$INTERACTIVE" ]] && helmfile_opt_flags+=("$INTERACTIVE")
-helmfile -f . -e service_cluster "${helmfile_opt_flags[@]}" apply --suppress-diff
+helmfile -f . -e service_cluster "${helmfile_opt_flags[@]}" sync --suppress-diff
```

Comment:
Why are you doing this change?

Reply:
This is the description of the commands:

    sync      sync all resources from state file (repos, releases and chart deps)
    apply     apply all resources from state file only when there are changes

So, i think it would be good for the sake of recovering a state? Not sure i have done changes in the right place. It was the only place i could find in where it would effect the "apply" command as the issuer wanted. Couldn't find a "helmfile apply" statement in either apps.sh or bootstrap.sh .

Reply:
I'm not sure if this actually will change something in practice, but if it does it's definitely worth mentioning in the changelog as this is changing the apply command, not dry-run.

Reply:
The reason why this was even brought up in the first place is that helm-diff does not check the acutally state in kubernetes, and thus if somebody has made any manual changes to let's say a deployment, helm-diff won't notice that and thus not upgrade the release. sync, will always upgrade the releases to a new version thus syncing the state on every single apply.

Personally, now that we are aware of the fact of this difference between apply and sync. Whether or not we want to run sync only and never apply, I can't say.

Reply:
Worth investigating before merging? :smile:

Reply:
I think we said that we should use sync since this is the same as running `helm upgrade --install` (instead of `helm apply` for when apply is used) which is how you would do it if you were using helm.

Reply:
`helm apply` isn't a thing I think? At least not on my end:
```
% helm apply
Error: unknown command "apply" for "helm"
Run 'helm --help' for usage.
% helm version --short
v3.2.4+g0ad800e
```

Reply:
Just did a quick search since I wasn't totally aware of the commands and I think the top answer [here](https://stackoverflow.com/questions/59703760/helmfile-sync-vs-helmfile-apply) explained it pretty clearly.

To me it feels like we should not run `helmfile sync` since it's going to basically upgrade the chars even though there are no changes made. Or am I missing something here?

Reply:
Oh, yea I misread the documentation there.
But I think the reason to why we want this is (as olle said) diff will not notice stuff that's manually edited. And since that might be the case that we need to do that (for an emergency fix or something) we should not allow this manual change to hang around for a couple of upgrades until the release are actually redeployed and breaks everything again (maybe months later). So in my mind it's better that we notice this change quicker when we might have a chance of remembering why. But there might be some issues with upgrading the release that I don't see?

Reply:
Gotcha! I'm a bit worried that manual changes that have been done would be destroyed by a sync though? For example, what happens if you have made manual changes to some labels and it runs an upgrade of the chart release? It's all of these unknowns that I would be a bit cautious introducing before we actually know more.

If anyone here actually have made some effort into testing this and researching sync vs apply implications please add it to the changelog so that we don't introduce this without anyone knowing what is going on.